katharsis-core/src/main/java/io/katharsis/legacy/queryParams/DefaultQueryParamsConverter.java

```diff
@@ -0,0 +1,138 @@
+package io.katharsis.legacy.queryParams;
+
+import java.math.BigDecimal;
+import java.util.*;
+
+import io.katharsis.core.internal.utils.StringUtils;
+import io.katharsis.legacy.queryParams.include.Inclusion;
+import io.katharsis.legacy.queryParams.params.*;
+import io.katharsis.queryspec.*;
+import io.katharsis.resource.information.ResourceInformation;
+import io.katharsis.resource.registry.RegistryEntry;
+import io.katharsis.resource.registry.ResourceRegistry;
+
+@SuppressWarnings({"deprecation"})
+public class DefaultQueryParamsConverter implements QueryParamsConverter {
+
+    private ResourceRegistry resourceRegistry;
+
+    public DefaultQueryParamsConverter(ResourceRegistry resourceRegistry) {
+        this.resourceRegistry = resourceRegistry;
+    }
+
+    @Override
+    public QueryParams fromParams(Class<?> rootType, QuerySpec querySpec) {
+        QueryParams params = new QueryParams();
+        applyIncludedFields(querySpec, params);
+        applySorting(querySpec, params);
+        applyRelatedFields(querySpec, params);
+        applyPaging(querySpec, params);
+        applyFiltering(querySpec, params);
+
+        params.setGrouping(new TypedParams<>(Collections.unmodifiableMap(new HashMap<String, GroupingParams>())));
+        return params;
+    }
+
+    private String getResourceType(Class<?> resourceClass) {
+        RegistryEntry registryEntry = resourceRegistry.getEntryForClass(resourceClass);
+        if (registryEntry == null) {
+            throw new IllegalArgumentException("resourceType for class " + resourceClass.getName() + " not found");
+        }
+        ResourceInformation resourceInformation = registryEntry.getResourceInformation();
+        return resourceInformation.getResourceType();
+    }
+
+    protected void applyFiltering(QuerySpec spec, QueryParams queryParams) {
```

Comment:
now that we are able to use java 8 I would really prefer to see this as a couple of lambdas.